### PR TITLE
add new cf params

### LINKED ISSUE #193: Customise BatchSize and StartingPosition of event triggers
**Describe the enhancement:**

**Describe a specific use case for the enhancement or feature:**

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## What does this PR do?
Add multiple customisation parameters for the lambda deployment.

## Why is it important?
We try to have the forwarder working OOTB for 90% of the scenarios, but still there are ones that need further customisation: exposing a few extra parameters will unlock that.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-
Closes #193 #187

### CODE REVIEW

**Comment by aspacca** [comment]
documentation update at https://github.com/elastic/observability-docs/pull/2473

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-elastic-apm>=6.12.0,<7.0
-ecs_logging>=2.0.0,<3
-boto3>=1.24.96,<2
-elasticsearch>=7.16,<7.17
-PyYAML>=6.0,<6.1
-aws_lambda_typing>=2.16.1,<3
-ujson>=5.5.0,<5.6
+elastic-apm==6.13.2
```
Comment (aspacca) [inline requirements.txt:1]: see my comment at https://github.com/elastic/elastic-serverless-forwarder/issues/192#issuecomment-1362431039

> The version published on the serverless repo is the same as the one tagged on 1.5 on github, but as you can see boto requirement is not pinned at a specific version (https://github.com/elastic/elastic-serverless-forwarder/blob/lambda-v1.5.0/requirements.txt#L3). this means the actual version used was resolved at the moment of publishing the on serverless repo. I've checked and it is v1.25.2
> 
> We'll change this behaviour in order to be able to identify the exact direct dependency version

File: requirements-tests.txt [hunk 2]
```diff
@@ -1,13 +1,14 @@
-mock>=4.0.3,<4.1
+mock>=4.0.3,<5.1
```
Comment (aspacca) [inline requirements-tests.txt:1]: from https://github.com/elastic/elastic-serverless-forwarder/pull/198

File: requirements-tests.txt [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 pytest>=7.1.3,<7.3
 pytest-cov>=4.0.0,<4.1
 pytest-benchmark>=3.4.1,<4.1
-coverage>=6.5.0,<6.6
-localstack[runtime]>=1.2.0,<1.3
-docker==5.0.0
+coverage>=6.5.0,<7.1
```
Comment (aspacca) [inline requirements-tests.txt:5]: from https://github.com/elastic/elastic-serverless-forwarder/pull/195

File: requirements-tests.txt [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+localstack[runtime]>=1.3.0,<1.4
```
Comment (aspacca) [inline requirements-tests.txt:6]: from https://github.com/elastic/elastic-serverless-forwarder/pull/189

File: .internal/aws/scripts/dist.sh [hunk 5]
```diff
@@ -78,7 +78,6 @@ sam publish --template "${TMPDIR}/.aws-sam/build/macro/packaged.yaml" --region "
 sam build --debug --use-container --build-dir "${TMPDIR}/.aws-sam/build/application" --template-file "${TMPDIR}/application.yaml" --region "${REGION}"
 sam package --template-file "${TMPDIR}/.aws-sam/build/application/template.yaml" --output-template-file "${TMPDIR}/.aws-sam/build/application/packaged.yaml" --s3-bucket "${BUCKET}" --region "${REGION}"
 sam publish --template "${TMPDIR}/.aws-sam/build/application/packaged.yaml" --region "${REGION}"
-aws s3 cp "${TMPDIR}/.aws-sam/build/application/packaged.yaml" "s3://${BUCKET}/application.yaml"
```
Comment (aspacca) [inline .internal/aws/scripts/dist.sh:81]: debug leftover from previous commit

**Review by girodav — APPROVED** [review]

File: publish_lambda.sh [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+CLONED_FOLDER="${TMPDIR}/sources"
+PACKAGE_FOLDER="${CLONED_FOLDER}/package"
+GIT_REPO="https://github.com/elastic/elastic-serverless-forwarder.git"
+
+trap 'rm -rf ${TMPDIR}' EXIT
+
+aws s3api get-bucket-location --bucket "${BUCKET}" || aws s3api create-bucket --acl private --bucket "${BUCKET}" --region "${REGION}" --create-bucket-configuration LocationConstraint="${REGION}"
+
+mkdir -v -p "${CLONED_FOLDER}"
+git clone --depth 1 --branch "${TAG_NAME}" "${GIT_REPO}" "${CLONED_FOLDER}"
```
Comment (girodav) [inline publish_lambda.sh:41]: What's the reasoning behind introducing `TAG_NAME` ? Aren't we always considering `main` as the source of truth?

**Review by girodav — APPROVED** [review]

**Pull request merged**